x/gamm/pool-models/stableswap/pool.go

```diff
@@ -24,8 +24,12 @@ var _ types.PoolI = &Pool{}
 func NewStableswapPool(poolId uint64, stableswapPoolParams PoolParams, initialLiquidity sdk.Coins, scalingFactors []uint64, futureGovernor string) (Pool, error) {
 	if len(scalingFactors) == 0 {
 		scalingFactors = []uint64{1, 1}
-	} else if scalingFactors[0] == 0 || scalingFactors[1] == 0 {
-		return Pool{}, types.ErrInvalidStableswapScalingFactors
+	}
+
+	for _, scalingFactor := range scalingFactors {
+		if scalingFactor == 0 || int64(scalingFactor) <= 0 {
+			return Pool{}, types.ErrInvalidStableswapScalingFactors
+		}
 	}
```

Comment:
This is reused, lets extract this to a function in stableswap for validate scaling factors

Reply:
Great point – extracted into `validateScalingFactors` function